torch/csrc/jit/pickler.cpp

```diff
@@ -525,60 +533,49 @@ void Unpickler::run() {
     if (opcode == OpCode::STOP) {
       return;
     }
-    last_opcode_ = opcode;
   }
 
   AT_ERROR("Overran buffer while unpickling data, didn't find STOP opcode");
 }
+void Unpickler::setInput(size_t memo_id) {
+  AT_ASSERT(!stack_.empty());
+  if (memo_id >= memo_table_.size()) {
+    memo_table_.insert(
+        memo_table_.end(), memo_id - memo_table_.size(), IValue());
+    memo_table_.push_back(stack_.back());
+  } else {
+    memo_table_[memo_id] = stack_.back();
+  }
+}
+
+template <typename T>
+static IValue toSpecializedList(const IValue& generic) {
```

Comment:
Is the extra copy this is now doing going to be a perf bottleneck later on that we'll end up reverting back to specializing when the list is created?

Reply:
So, before the patch I landed today, we were already accidentally doing this copy anyway. I have a plan to try to remove specialization entirely that I am going to try. If it end ups being a perf bottleneck, I can restore the copy-free version, but it is a lot of confusing logic so I'd like to avoid it if it is not needed.